network/f5/bigip_virtual_server.py

```diff
@@ -0,0 +1,405 @@
+    all_profiles:
+        description:
+            - "List of all Profiles (HTTP,ClientSSL,ServerSSL,etc) that must be used by the virtual server"
+        required: false
+        default: null
+    pool:
+        description:
+            - "Default pool for the virtual server"
+        required: false
+        default: null
+    snat:
+        description:
+            - "Source network address policy"
+        required: false
+        default: None
+
+    description:
+        description:
+            - "Virtual server description."
+        required: false
+        default: null
+'''
+
+EXAMPLES = '''
+
+## playbook task examples:
+
+---
+# file bigip-test.yml
+# ...
+  - name: Add VS
+    local_action:
+        module: bigip_virtual_server
+        server: lb.mydomain.net
+        user: admin
+        password: secret
+        state: present
+        partition: MyPartition
+        name: myvirtualserver
+        destination: "{{ ansible_default_ipv4["address"] }}"
+        port: 443
+        pool: "{{ mypool }}"
+        snat: Automap
+        description: Test Virtual Server
+        all_profiles:
+            - http
+            - clientssl
+
+  - name: Modify Port of the Virtual Server
+    local_action:
+        module: bigip_virtual_server
+        server: lb.mydomain.net
+        user: admin
+        password: secret
+        state: present
+        partition: MyPartition
+        name: myvirtualserver
+        port: 8080
+
+  - name: Delete pool
+    local_action:
+        module: bigip_virtual_server
+        server: lb.mydomain.net
+        user: admin
+        password: secret
+        state: absent
+        partition: MyPartition
+        name: myvirtualserver
+'''
+
+
+# ==========================
+# bigip_virtual_server module specific
+#
+
+# map of state values
+STATES={'enabled': 'STATE_ENABLED',
+        'disabled': 'STATE_DISABLED'}
+STATUSES={'enabled': 'SESSION_STATUS_ENABLED',
+          'disabled': 'SESSION_STATUS_DISABLED',
+          'offline': 'SESSION_STATUS_FORCED_DISABLED'}
+
+def vs_exists(api, vs):
+    # hack to determine if pool exists
+    result = False
+    try:
+        api.LocalLB.VirtualServer.get_object_status(virtual_servers=[vs])
+        result = True
+    except bigsuds.OperationFailed, e:
+        if "was not found" in str(e):
+            result = False
+        else:
+            # genuine exception
+            raise
+    return result
+
+def vs_create(api,name,destination,port,pool):
+    _profiles=[[{'profile_context': 'PROFILE_CONTEXT_TYPE_ALL', 'profile_name': 'tcp'}]]
+    try:
+        api.LocalLB.VirtualServer.create(
+            definitions = [{'name': [name], 'address': [destination], 'port': port, 'protocol': 'PROTOCOL_TCP'}],
+            wildmasks = ['255.255.255.255'],
+            resources = [{'type': 'RESOURCE_TYPE_POOL', 'default_pool_name': pool}],
+            profiles = _profiles)
+        result = True 
+        desc = 0
+    except Exception, e :
+        print e.args
```

Comment:
Don't use print in modules, use either exit or fail json functions.  Also the point of having localized exceptions is to add an error message on top that is more helpful than most library exception messages + traceback (but we also want to include those), if you are just going to pass the exception message, there is really no difference with using the general catchall exception.